pkg/model/enum_def.go

```diff
@@ -30,29 +29,15 @@ type EnumValue struct {
 // either a CRD or a TypeDef
 type EnumDef struct {
 	Names  names.Names
-	GoType string
 	Values []EnumValue
 }
 
-func NewEnumDef(names names.Names, goType string, values []interface{}) (*EnumDef, error) {
+func NewEnumDef(names names.Names, values []string) (*EnumDef, error) {
 	enumVals := make([]EnumValue, len(values))
 	for x, item := range values {
-		if goType == "string" {
```

Comment:
Note to reviewers: there are only string enums in the AWS service API model definitions. Thus, I removed the GoType stuff here and simplified things.